docs/examples/Example_data_fusion_from_sensors.py

```diff
@@ -0,0 +1,516 @@
+
+# Loaded the plotter for the various metrics.
+from stonesoup.plotter import MetricPlotter
+
+# %%
+# Set up the metrics
+# ^^^^^^^^^^^^^^^^^^
+
+# load the metrics for the filters
+basic_UKF = BasicMetrics(generator_name='Unscented Kalman Filter', tracks_key='UKF_tracks',
+                         truths_key='truths')
+basic_EKF = BasicMetrics(generator_name='Extended Kalman Filter', tracks_key='EKF_tracks',
+                         truths_key='truths')
+basic_PF = BasicMetrics(generator_name='Particle Filter', tracks_key='PF_tracks',
+                        truths_key='truths')
+
+# OSPA
+ospa_UKF_truth = OSPAMetric(c=40, p=1, generator_name='OSPA_UKF_truths',
+                            tracks_key='UKF_tracks',  truths_key='truths')
+ospa_EKF_truth = OSPAMetric(c=40, p=1, generator_name='OSPA_EKF_truths',
+                            tracks_key='EKF_tracks',  truths_key='truths')
+ospa_PF_truth = OSPAMetric(c=40, p=1, generator_name='OSPA_PF_truths',
+                           tracks_key='PF_tracks',  truths_key='truths')
+
+# Use the track associator
+associator = TrackToTruth(association_threshold=30)
+
+# Use a metric manager to deal with the various metrics
+metric_manager = MultiManager([basic_UKF,
+                               basic_EKF,
+                               basic_PF,
+                               ospa_UKF_truth,
+                               ospa_EKF_truth,
+                               ospa_PF_truth],
+                              associator)
+
+# %%
+# Run simulations
+# ^^^^^^^^^^^^^^^
+
+# define the Detections from the two sensors
+s1_detections = []
+s2_detections = []
+radar_path = []
+
+# use the generator function from the simulators
+g1 = radar_simulator1.detections_gen()
+g2 = radar_simulator2.detections_gen()
+
+# Create the tracks
+ukf_tracks = set()
+ekf_tracks = set()
+pf_tracks = set()
+truths = set()
+
+# list for all detections
+full_detections = []
+
+for t in range(number_of_steps):  # loop over the various time-steps
+    detections_1 = next(g1)
+    s1_detections.extend(detections_1[1])
+    detections_2 = next(g2)
+    s2_detections.extend(detections_2[1])
+    full_detections.extend(detections_1[1])
+    full_detections.extend(detections_2[1])
+
+    for detections in [detections_1, detections_2]:
+
+        # Run the Unscented Kalman tracker
+        UKF_tracker.detector = DummyDetector(current=detections)
+        UKF_tracker.__iter__()
+        _, tracks = next(UKF_tracker)
+        ukf_tracks.update(tracks)
+        del tracks
+
+        # Run the Extended Kalman filter
+        EKF_tracker.detector = DummyDetector(current=detections)
+        EKF_tracker.__iter__()
+        _, tracks = next(EKF_tracker)
+        ekf_tracks.update(tracks)
+        del tracks
+
+        # Run the Particle filter Tracker
+        PF_tracker.detector = DummyDetector(current=detections)
+        PF_tracker.__iter__()
+        _, tracks = next(PF_tracker)
+        pf_tracks.update(tracks)
```

Comment:
Calls to `__iter__`  and `DummyDetector` aren't the best way to approach this. Please see [latest *Multi-Sensor Fusion: Covariance Intersection Using Tracks as Measurements* example](https://stonesoup.readthedocs.io/en/v1.2/auto_examples/Track2Track_Fusion_Example.html#sphx-glr-auto-examples-track2track-fusion-example-py), and #882

Reply:
I've amended the part of the code where  `__iter__` and `DummyDetector` were used.